### PR TITLE
optimize docker and select node version

### PR DESCRIPTION
**Pull request by cesarjorgemartinez**
We do not have a section for docker docs (in installationguide).
Which text can be added?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,12 +7,62 @@
...
+  /bin/cp -f /usr/lib/locale/locale-archive /usr/lib/locale/locale-archive.tmpl && \
+  build-locale-archive && \
+  find /opt/iotajson -name '.[^.]*' 2>/dev/null | xargs -r rm -rf && \
+  # Clean npm cache
+  npm cache clean && \
+  # Don't need unused files inside docker images
+  rm -rf /tmp/* /usr/local/lib/node_modules/npm/man /usr/local/lib/node_modules/npm/doc /usr/local/lib/node_modules/npm/html && \
+  # We don't need to manage Linux account passwords requisites: lenght, mays/mins, etc
+  # This cannot be removed using yum as yum uses hard dependencies and doing so will uninstall essential packages
+  rm -rf /usr/share/cracklib && \
```
Comment (AlvaroVega) [inline Dockerfile]: I'm not sure if cracklib is used by this node component.

**Comment by AlvaroVega** [comment]
I'm trying build this image with an especific node version:
```
 sudo docker build -t iotagent-json --build-arg NODEJS_VERSION=0.10.26 .
```
but I'm not able:
```
Sending build context to Docker daemon 868.4 kB
Step 1 : FROM centos:6
 ---> f07f6ca555a5
Step 2 : RUN yum update -y && yum install -y wget   && wget http://dl.fedoraproject.org/pub/epel/6/x86_64/epel-release-6-8.noarch.rpm && yum localinstall -y --nogpgcheck epel-release-6-8.noarch.rpm   && yum install -y npm git
 ---> Using cache
 ---> 4335ed6cfd3a
Step 3 : COPY . /opt/iotajson
 ---> Using cache
 ---> bdbd9806dd31
Step 4 : WORKDIR /opt/iotajson
 ---> Using cache
 ---> 5d43226d3057
Step 5 : RUN npm install
 ---> Using cache
 ---> fd361bf215b1
Step 6 : ENTRYPOINT bin/iotagent-json config.js
 ---> Using cache
 ---> cbf117d3938f
One or more build-args [NODEJS_VERSION] were not consumed, failing build.
```
What's wrong?

File: Dockerfile (hunk 1 above)
Comment (cesarjorgemartinez) [inline Dockerfile]: This component not use cracklib

Comment (cesarjorgemartinez) [inline Dockerfile]: This component not use cracklib

**Comment by cesarjorgemartinez** [comment]
sudo docker build --build-arg NODEJS_VERSION=0.10.46 -t iotagent-json -f Dockerfile .

**Comment by AlvaroVega** [comment]
LGTM

**Pull request merged**